src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -33,75 +36,938 @@ const ActorInit En_In_InitVars = {
+    }
+}
+
+void func_80A7AE84(EnIn* this, GlobalContext* globalCtx) {
+    Gameplay_ChangeCameraStatus(globalCtx, this->activeCamId, 7);
+    Gameplay_ClearCamera(globalCtx, this->camId);
+    func_8002DF54(globalCtx, &this->actor, 7);
+    Interface_ChangeAlpha(0x32);
+    this->actionFunc = func_80A7AEF0;
+}
+
+void func_80A7AEF0(EnIn* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 yaw;
+    Vec3f pos = this->actor.posRot.pos;
+
+    pos.x += 90.0f * Math_SinS(this->actor.shape.rot.y);
+    pos.z += 90.0f * Math_CosS(this->actor.shape.rot.y);
+    yaw = Math_Vec3f_Yaw(&pos, &player->actor.posRot.pos);
+    if (ABS(yaw) > 0x4000) {
+        globalCtx->nextEntranceIndex = 0x0476;
+        globalCtx->sceneLoadFlag = 0x14;
+        globalCtx->fadeTransition = 5;
+        this->actionFunc = func_80A7B018;
+    } else if (this->unk_308.unk_00 == 2) {
+        globalCtx->msgCtx.unk_E3E7 = 4;
+        globalCtx->msgCtx.msgMode = 0x36;
+        this->unk_308.unk_00 = 0;
+    }
+}
+
+void func_80A7B018(EnIn* this, GlobalContext* globalCtx) {
+}
+
+void func_80A7B024(EnIn* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (player->rideActor != NULL) {
+        player->rideActor->freezeTimer = 10;
+    }
+    player->actor.freezeTimer = 10;
+    if (this->unk_308.unk_00 == 2) {
+        if (1) {}
+        if (!(gSaveContext.eventChkInf[1] & 0x800) && (gSaveContext.infTable[10] & 0x800)) {
+            gSaveContext.eventChkInf[1] |= 0x800;
+            gSaveContext.infTable[10] |= 0x800;
+        }
+        func_80A79BAC(this, globalCtx, 0, 0x26);
+        gSaveContext.eventInf[0] = gSaveContext.eventInf[0] & ~0xF;
+        gSaveContext.eventInf[0] = (gSaveContext.eventInf[0] & ~0x8000) | 0x8000;
+        globalCtx->msgCtx.unk_E3E7 = 4;
+        globalCtx->msgCtx.msgMode = 0x36;
+        this->unk_308.unk_00 = 0;
+    }
+}
+
+void EnIn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ColliderCylinder* collider;
+    EnIn* this = THIS;
+
+    if (this->actionFunc == func_80A79FB0) {
+        this->actionFunc(this, globalCtx);
+        return;
+    }
+    collider = &this->collider;
+    Collider_CylinderUpdate(&this->actor, collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &collider->base);
+    if (this->actionFunc != func_80A7A304) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+        if (this->skelAnime.animation == &D_06001BE0 && ((gSaveContext.eventInf[0] & 0xF) != 6)) {
+            func_80A79690(&this->skelAnime, this, globalCtx);
+        }
+        func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    }
+    func_80A79A2C(this);
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc != func_80A7A304) {
+        func_80A79AB4(this, globalCtx);
+        if (gSaveContext.timer2Value < 6 && gSaveContext.timer2State != 0 && this->unk_308.unk_00 == 0) {
+            if (func_8002F194(&this->actor, globalCtx)) {}
```

Comment:
Does this have to be in a `if` condition?

Reply:
It does unfortunately, in this case it really does generate an extra branch (likely)